src/aws_encryption_sdk/__init__.py

```diff
@@ -158,27 +170,38 @@ def decrypt(self, **kwargs):
 
         .. code:: python
 
+            >>> import boto3
+            >>> from aws_cryptographic_materialproviders.mpl import AwsCryptographicMaterialProviders
+            >>> from aws_cryptographic_materialproviders.mpl.config import MaterialProvidersConfig
+            >>> from aws_cryptographic_materialproviders.mpl.models import CreateAwsKmsKeyringInput
+            >>> from aws_cryptographic_materialproviders.mpl.references import IKeyring
             >>> import aws_encryption_sdk
             >>> client = aws_encryption_sdk.EncryptionSDKClient()
-            >>> kms_key_provider = aws_encryption_sdk.StrictAwsKmsMasterKeyProvider(key_ids=[
-            ...     'arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
-            ...     'arn:aws:kms:us-east-1:3333333333333:key/33333333-3333-3333-3333-333333333333'
-            ... ])
-            >>> my_ciphertext, encryptor_header = client.decrypt(
+            >>> mat_prov: AwsCryptographicMaterialProviders = AwsCryptographicMaterialProviders(
+            ...     config=MaterialProvidersConfig()
+            ... )
+            >>> keyring_input: CreateAwsKmsKeyringInput = CreateAwsKmsKeyringInput(
+            ...     kms_key_id='arn:aws:kms:us-east-1:2222222222222:key/22222222-2222-2222-2222-222222222222',
+            ...     kms_client=boto3.client('kms', region_name="us-west-2")
+            ... )
+            >>> kms_keyring: IKeyring = mat_prov.create_aws_kms_keyring(
+            ...     input=keyring_input
+            ... )
+            >>> my_plaintext, decryptor_header = client.decrypt(
             ...     source=my_ciphertext,
-            ...     key_provider=kms_key_provider
+            ...     keyring=kms_keyring
             ... )
 
         :param config: Client configuration object (config or individual parameters required)
         :type config: aws_encryption_sdk.streaming_client.DecryptorConfig
         :param source: Source data to encrypt or decrypt
         :type source: str, bytes, io.IOBase, or file
         :param materials_manager: `CryptoMaterialsManager` that returns cryptographic materials
-            (requires either `materials_manager` or `key_provider`)
+            (requires either `materials_manager` or `keyring`)
         :type materials_manager: aws_encryption_sdk.materials_managers.base.CryptoMaterialsManager
-        :param key_provider: `MasterKeyProvider` that returns data keys for decryption
```

Comment:
Don't remove documentation for key_provider as a possible arg. We aren't removing the arg, just making it a "less-preferred" input in examples.